chains/evm/deployment/v1_7_0/changesets/configure_chain_for_lanes.go

```diff
@@ -0,0 +1,156 @@
+package changesets
+
+import (
+	"fmt"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/deployment/utils/changesets"
+	datastore_utils "github.com/smartcontractkit/chainlink-ccip/chains/evm/deployment/utils/datastore"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/deployment/v1_2_0/operations/router"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/deployment/v1_7_0/operations/ccv_aggregator"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/deployment/v1_7_0/operations/ccv_proxy"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/deployment/v1_7_0/operations/commit_onramp"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/deployment/v1_7_0/operations/fee_quoter_v2"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/deployment/v1_7_0/sequences"
+	"github.com/smartcontractkit/chainlink-deployments-framework/chain/evm"
+	"github.com/smartcontractkit/chainlink-deployments-framework/datastore"
+	cldf_deployment "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+)
+
+type RemoteChainConfig struct {
+	AllowTrafficFrom            bool
+	CCIPMessageSource           datastore.AddressRef
+	DefaultCCVOffRamps          []datastore.AddressRef
+	LaneMandatedCCVOffRamps     []datastore.AddressRef
+	DefaultCCVOnRamps           []datastore.AddressRef
+	LaneMandatedCCVOnRamps      []datastore.AddressRef
+	DefaultExecutor             datastore.AddressRef
+	CommitOnRampDestChainConfig sequences.CommitOnRampDestChainConfig
+	FeeQuoterDestChainConfig    fee_quoter_v2.DestChainConfig
+}
+
+type ConfigureChainForLanesCfg struct {
+	ChainSel     uint64
+	RemoteChains map[uint64]RemoteChainConfig
+}
+
+func (c ConfigureChainForLanesCfg) ChainSelector() uint64 {
+	return c.ChainSel
+}
+
+var ConfigureChainForLanes = changesets.NewFromOnChainSequence(changesets.NewFromOnChainSequenceParams[
+	sequences.ConfigureChainForLanesInput,
+	evm.Chain,
+	ConfigureChainForLanesCfg,
+]{
+	Sequence: sequences.ConfigureChainForLanes,
+	Describe: func(in sequences.ConfigureChainForLanesInput, dep evm.Chain) string {
+		remoteChains := make([]uint64, 0, len(in.RemoteChains))
+		for chainSel := range in.RemoteChains {
+			remoteChains = append(remoteChains, chainSel)
+		}
+		return fmt.Sprintf("Configure remote chain connnections on %s: %v", dep, remoteChains)
+	},
+	ResolveInput: func(e cldf_deployment.Environment, cfg ConfigureChainForLanesCfg) (sequences.ConfigureChainForLanesInput, error) {
+		staticAddrs, err := datastore_utils.FindAndFormatEachRef(e.DataStore, []datastore.AddressRef{
```

Comment:
I like the overall format of the NewFromOnChainSequence, feels like the Resolve methods bloads the readability of the changesets to get to the crux of the logic.
could we go back to having separate methods like toSeqInput or something of that sorts?